### ISSUE TITLE
File mode specification error: (wrong-type-argument (integer 1 65535) lsp-intellij-server-port port)

### ISSUE THREAD

**Issue by prabhu-madanagopal** [Ruin0x11/lsp-intellij#9]
Hi, 

I have the server running per instructions from here: https://github.com/Ruin0x11/intellij-lsp-server

I have installed lsp-intellij from Melpa and followed the setup instructions. 

However when I open any java file from the project in IDEA, I get this error:

File mode specification error: (wrong-type-argument (integer 1 65535) lsp-intellij-server-port port)

Could you please help resolve this issue?

**Comment by jonasws** [comment]
I have encountered this issue as well. It appears that referencing the `lsp-intellij-server-port` variable is done wrong at https://github.com/Ruin0x11/lsp-intellij/blob/master/lsp-intellij.el#L569. It works for me locally if I remove the leading quote, making it
```lisp
(lsp-define-tcp-client lsp-intellij "intellij" #'lsp-intellij--get-root lsp-intellij-dummy-executable
                       "127.0.0.1" lsp-intellij-server-port
:initialize #'lsp-intellij--initialize-client)
```

@Ruin0x11 Can you look into to confirming this behavior, and perhaps commit a fix, if this is indeed the remedy to the issue? I'm not completely sure if I'm missing something here.

**Comment by prabhu-madanagopal** [comment]
Thanks jonasws. I found the same, removing the single quote resolve the issue. I am able to lsp-intellij. It works, but the server becomes unresponsive most time because it is a very large project.

**Comment by Ruin0x11** [comment]
Okay, I'll look into this.

Parts of the server like searching for references block the server due to
issues with IntelliJ's thread model causing deadlocks.

2018年8月27日(月) 11:51 prabhu-madanagopal <notifications@github.com>:

**Comment by prabhu-madanagopal** [comment]
Ruin0x11, thanks for your work on IntelliJ lsp.

I was using lsp-java which uses eclipse.jdt.ls but I had to clean up workspace before each startup because the language server does not successfully save the workspace. Each time it took ~30 min for eclipse language server to startup. But the references and rename work after startup.

lsp-intellij startup has no issues, IntelliJ language server startups within a minute. However the references timing out because of deadlock is a blocker. Is there a way to resolve this?

Thanks, Prabhu

**Comment by prabhu-madanagopal** [comment]
Never mind, I can work with it by increasing the lsp-response-timeout to 2 minutes.

**Comment by Ruin0x11** [comment]
It should be fixed in cf30f0ac63bd0140e758840b8ab070e8313697b2.
